### DISCUSSION TITLE
Data disappearing

### DISCUSSION THREAD

**Discussion by PerpSearch** [benjaminjonard/koillection#1096] [Q&A] [answered]
I created a collection of items that included 10 data fields that included an image and a $ value.  It took several hours to enter 328 records into the database.  Occasionally I had to go back and enter some I skipped, but once done it felt pretty good to move from the spreadsheet to Koillection.  

I know everything was entered after a while because I went back and checked for missing data points against my spreadsheet.  A few hours later I happened to check and found 9 records that had data missing.  There seemed to be no commonality in the records, but as I re-entered the data I was able to confirm that it was there before when I had assumed it was my oversight.  It wasn't.  

The log has 41 lines similar to the following:

**[notice] Migration App\Migrations\Mysql\Version20230525100104 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."**

and one of these 

**[24-Mar-2024 19:22:36] NOTICE: PHP message: PHP Warning:  Can't preload unlinked class Zenstruck\Foundry\Factory@anonymous: Unknown parent Zenstruck\Foundry\Factory in /var/www/koillection/vendor/zenstruck/foundry/src/functions.php on line 45**

 if that's of any help.

I'm running this in a docker container on a Synology DS723+  having used the tutorial provided [here](https://mariushosting.com/how-to-install-koillection-on-your-synology-nas/).  I do like the program, but I worry about it's dependability now.  I'm willing to provide the complete log(s) if it's be helpful.
Thanks.

**Comment by benjaminjonard** [comment]
Which version of Koillection do you use ? 
Alse when you first added the missing data, did you apply a template or change the fields positions ? 

The two logs messages can be safely ignored. I need to find a way to hide them as they appear for everyone and are irrelevant

**Reply by PerpSearch** [reply]
> Alse when you first added the missing data, did you apply a template or change the fields positions ?

Oops, sorry I missed this question when I fixated on the version issue.  Yes, I applied a template with choices populated, but I used the same template for all entries.

Also, I'm willing to share access to it if it would help.

**Reply by benjaminjonard** [reply]
Would you mind sharing your template and 1 item where the data disappeared ?

**Reply by PerpSearch** [reply]
I had to leave for a bit, but when I get back to the house we'll work something out. Thanks again.

**Reply by PerpSearch** [reply]
I am home for the day and able to provide you with whatever you need, but you'll have to tell me how.  

While you have me thinking about it, it's possible the 9 data items that disappeared may have happed when I edited the collection to change the view from grid to list.  That was when I first noticed it anyway.  When changing between those views, I had gotten a 500 Internal Server Error a couple times, but it seemed to work anyway.

**Reply by benjaminjonard** [reply]
Ok thanks for your time. Could you share a screenshot of the template you use and one of an item that lost a data field ?

**Reply by PerpSearch** [reply]
I checked the collection in the list format and the data point for "mat shape" was populated. Now it's missing.

![2024-04-03_163716](https://github.com/benjaminjonard/koillection/assets/92053831/7aeaf421-9edb-4a28-860d-725a2d654c82)
![2024-04-03_163351](https://github.com/benjaminjonard/koillection/assets/92053831/75d4c931-c776-41d6-8fe2-fdf25df17aca)

**Reply by benjaminjonard** [reply]
Thanks, if you could also provide a screenshot of the "Mat shape" choice list that would be great. I'll try to reproduce the issue tomorrow

**Reply by benjaminjonard** [reply]
Ok I may have found it. Have you made any change in the choice list ?  I changed one entry in my list and it disappeared from my item like you. (the good news would be that the data is still present in the database)

**Reply by PerpSearch** [reply]
Yes I have done that, perhaps a couple of times.  I'm actually still experimenting with the program features and settings so yeah. The good news is that you've actually been able to replicate it resolve the issue or tell me what to do differently.  Either way it's a win.
Cool.
Also, data loss is easily overcome by reentering it cuz there's not much there and I'm kind a done for a while, but a fix down the road would be great.

**Reply by benjaminjonard** [reply, marked as answer]
Yeah I fixed the issue, it will ship with the next release. Thank you very much for your help

**Reply by PerpSearch** [reply]
No, thank you for all you've done here.  Later on we can talk about custom queries and reports ;) You have a link of some sort so I can buy you a coffee?

**Reply by benjaminjonard** [reply]
I haven't set up any links like that, but thanks for considering it, I appreciate it :+1:

**Reply by PerpSearch** [reply]
I just updated to the new version and after I noticed 5 records out of 335 were missing data.   The data fields that were zeroed out were all originally from a primary template.  I've not noticed any commonality with the records that were affected as yet, but I'm certain they were there before.  This is a small number of records, but I do back up the docker container twice a day to ensure I don't lose too much. Let me know what I can provide to help troubleshoot.

**Reply by benjaminjonard** [reply]
Well this is getting weird :sweat_smile: the last update had nothing to do with data fields. 
Are those data fields choice lists again ?

**Reply by PerpSearch** [reply]
Both choice lists and ones that have text entries.  The original image and all the tags were still there though. 

 A couple of days ago something occurred and I got a critical error with a return to home link. I would click that and return to the main page, but I  couldn't open the collection without getting that same error.  

Strange though because I could still search records and add records and edit them, but I was in a loop just trying to open the collection from the main page.  I restored a version from a couple days prior and we were good to go until last night with the update. 

I have been doing a lot of editing and I was checking frequently to see if anything was disappearing, but I thought we were in pretty good shape as everything was there . . . until it wasn't.

I'm going to experiment with it a little to see if I can cause some more data loss.

**Reply by PerpSearch** [reply]
BTW, here's the log file
**** 1/11 - Make sure /uploads folders exist ****
**** 2/11 - Create the symbolic link for the /uploads folder ****
**** 3/11 - Setting env variables ****
**** 4/11 - Migrate the database ****
[notice] Migrating up to App\Migrations\Postgresql\Version20240411083531
[notice] Migration App\Migrations\Mysql\Version20200429093452 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20200727114843 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20200812160901 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20200923193955 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20201010121132 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20201215214207 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20210127105342 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20210514100228 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20211207185256 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220206133834 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220206175335 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220208094335 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220630100634 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220701175209 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220713152238 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220716201458 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220717140437 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220722163001 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220722192217 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220725204054 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220829085551 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220909124604 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220916162932 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220917151134 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220918183100 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220920150306 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20220922183351 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20230317131927 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20230425131232 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20230519201736 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20230525100104 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20230613175623 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20230726074848 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20230801111844 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20230801130420 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20230803153034 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20231124105513 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20240105142108 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20240119000437 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20240204172429 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20240205152905 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20240329230249 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] Migration App\Migrations\Mysql\Version20240411083658 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."
[notice] finished in 4309.9ms, used 14M memory, 44 migrations executed, 1 sql queries

 [OK] Successfully migrated to version:                                         

      App\Migrations\Postgresql\Version20240411083531                           

**** 5/11 - Refresh cached values ****
Refreshing cached values for collections...
Refreshing cached values for wishlists...
Refreshing cached values for albums...
Done!
**** 6/11 - Create API keys ****
 [OK] Done!                                                                     
**** 7/11 - Create user and use PUID/PGID ****
-e  	User UID :	1026
-e  	User GID :	100
**** 8/11 - Set Permissions ****
**** 9/11 - Create nginx log files ****
**** 10/11 - Create symfony log files ****
**** 11/11 - Setup complete, starting the server. ****
[08-May-2024 00:33:53] NOTICE: PHP message: PHP Warning:  Can't preload unlinked class Zenstruck\Foundry\Factory@anonymous: Unknown parent Zenstruck\Foundry\Factory in /var/www/koillection/vendor/zenstruck/foundry/src/functions.php on line 45

**Reply by PerpSearch** [reply]
I could not replicate the "data loss" during my testing and I added 7 records, edited 180, edited the collection, added additional info, and renamed fields. Generally screwed around with it for several hours, I re-pulled the image and restarted the container. 
No loss of data.  I guess we're probably good and I'll monitor on my end and advise if I see anything.
Whatever was going on it seems to be fine now so lets just chalk this up to an anomaly or maybe just a "me" problem.

**Comment by PerpSearch** [comment]
I see you did it and it looks great!  I was just updating a few items and saw the update.  Now I just have to change 6 fields in 330 records and find someone to share it with. Thanks again for dong this.
